src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -649,8 +671,26 @@ static ucs_status_t uct_ib_mlx5_devx_md_open(struct ibv_device *ibv_device,
         goto err_free;
     }
 
-    if (UCT_IB_MLX5DV_GET(cmd_hca_cap, cap, dct)) {
-        dev->flags |= UCT_IB_DEVICE_FLAG_DC;
+    status = uct_ib_mlx5_devx_query_lag(md, &lag_state);
+    if (status != UCS_OK) {
```

Comment:
if query_lag failed need to fail md_open
if (status != UCS_OK) goto err_free;

Reply:
@yosefe But for older FW/device which query lag was not supported, the failure is reasonable, we should not treat it as md_open failure, but just fall back to the previours implementation.  Otherwise the older FW/device will always be failed. Could you please clarify this?

Reply:
ok, but also - is there a capability bit we can check to see if QUERY_LAG command is supported?

Reply:
@yosefe As decribed in PRM, lag_master could be the right bit. But I queried this bit and found on the BF2 its value for bond device is 0. This is not right. I'm discussing with FW guys.

Reply:
@yosefe According to the discussion with FW guys, lag_master is not used for this. It only indicates the right of changing lag(CREATE/MODIFY/DESTROY LAG).
I don't find any other bit can be used for this. Maybe we can check this with the syndrome id in the output of query result or the return values of mlx5dv_devx_general_cmd(). But seems this will be much complex to make sure the syndrome id or return id for unsupported op code.

Reply:
ok, so what do you think is the best method?

Reply:
@yosefe IMO, the current implementation is ok. Maybe we need to change "ucs_error("mlx5dv_devx_general_cmd(QUERY_LAG) failed: %m");" to ucs_debug to avoid error assert for older FW versions. I don't see critical case for QUERY_LAG failed and has to go to  err_free.

Reply:
ok